packages/devtools_app/test/test_infra/scenes/standalone_ui/property_editor_sidebar.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:devtools_app_shared/service.dart';
+import 'package:devtools_app_shared/shared.dart';
+import 'package:devtools_app_shared/ui.dart';
+import 'package:devtools_app_shared/utils.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:dtd/dtd.dart';
+import 'package:flutter/material.dart';
+import 'package:stager/stager.dart';
+
+import 'editor_service/simulated_editor.dart';
+import 'mock_editor_widget.dart';
+import 'utils.dart';
+
+/// To run, use the "standalone_ui/property_editor_sidebar" launch configuration with the
+/// `devtools/packages/` folder open in VS Code, or run:
+///
+/// flutter run -t test/test_infra/scenes/standalone_ui/property_editor_sidebar.stager_app.g.dart -d chrome
+class PropertyEditorSidebarScene extends Scene {
+  @override
+  Widget build(BuildContext context) {
+    return const _PropertyEditorSidebar();
+  }
+
+  @override
+  String get title => '$PropertyEditorSidebarScene';
+
+  @override
+  Future<void> setUp() async {
+    setStagerMode();
+    setGlobal(
+      DevToolsEnvironmentParameters,
+      ExternalDevToolsEnvironmentParameters(),
+    );
+    setGlobal(DTDManager, MockDTDManager());
+    setGlobal(IdeTheme, IdeTheme());
+    setGlobal(PreferencesController, PreferencesController());
+  }
+}
+
+class _PropertyEditorSidebar extends StatefulWidget {
+  const _PropertyEditorSidebar();
+
+  @override
+  State<_PropertyEditorSidebar> createState() => _PropertyEditorState();
+}
+
+class _PropertyEditorState extends State<_PropertyEditorSidebar> {
+  Stream<String>? clientLog;
+  DartToolingDaemon? clientDtd;
+  SimulatedEditor? editor;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: themeFor(
+        isDarkTheme: false,
+        ideTheme: _ideTheme(const VsCodeTheme.light()),
+        theme: ThemeData(useMaterial3: true, colorScheme: lightColorScheme),
+      ),
+      darkTheme: themeFor(
+        isDarkTheme: true,
+        ideTheme: _ideTheme(const VsCodeTheme.dark()),
+        theme: ThemeData(useMaterial3: true, colorScheme: darkColorScheme),
+      ),
+      home: Scaffold(
+        body:
+            clientLog != null && clientDtd != null && editor != null
+                ? MockEditorWidget(
+                  editor: editor!,
+                  clientLog: clientLog!,
+                  child: PropertyEditorSidebarPanel(clientDtd!),
+                )
+                : _DtdUriForm(
+                  onSaved: _connectToDtd,
+                  formKey: GlobalKey<FormState>(),
+                ),
+      ),
+    );
+  }
+
+  /// Creates an [IdeTheme] using the colours from the mock editor.
+  IdeTheme _ideTheme(VsCodeTheme vsCodeTheme) {
+    return IdeTheme(
+      backgroundColor: vsCodeTheme.editorBackgroundColor,
+      foregroundColor: vsCodeTheme.foregroundColor,
+      embedMode: EmbedMode.embedOne,
+    );
+  }
```

Comment:
these are duplicated between this file and `editor_sidebar.dart`. Can we share this code in some way?

Reply:
Pulled them into `/shared/common_ui.dart`